### PR TITLE
Remove `dot2hoa`

### PR DESCRIPTION
**Pull request by marcofavorito**
- remove dot2hoa script. It has been decided dot2hoa is not going to be supported soon. So rename the scripts such that it is clear 'hoa2dot' is the main entrypoint.

-  add hashing to LabelExpression and fix minor issues. 

-  add tests for parsing all the examples.

### CODE REVIEW

File: hoa2dot/core.py [hunk 1]
```diff
@@ -262,6 +265,10 @@ def __eq__(self, other):
         """Check equality between two AndLabelExpressions."""
         return isinstance(other, AndLabelExpression) and self.subexpressions == other.subexpressions
 
+    def __hash__(self):
+        """Compute the hash of a AndLabelExpression."""
```
Comment (francescofuggitti) [inline hoa2dot/core.py]: shouldn't the docstring be `an AndLabelExpression.` instead of  `a AndLabelExpression.`??

File: hoa2dot/core.py [hunk 2]
```diff
@@ -292,6 +299,10 @@ def __eq__(self, other):
         """Check equality between two OrLabelExpressions."""
         return isinstance(other, OrLabelExpression) and self.subexpressions == other.subexpressions
 
+    def __hash__(self):
+        """Compute the hash of a OrLabelExpression."""
```
Comment (francescofuggitti) [inline hoa2dot/core.py]: the same here. see above

File: hoa2dot/core.py [hunk 3]
```diff
@@ -317,6 +328,10 @@ def __eq__(self, other):
         """Check equality between two AtomLabelExpressions."""
         return isinstance(other, AtomLabelExpression) and self.propositions == other.propositions
 
+    def __hash__(self):
+        """Compute the hash of a AtomLabelExpression."""
```
Comment (francescofuggitti) [inline hoa2dot/core.py]: also here

File: hoa2dot/core.py [hunk 4]
```diff
@@ -386,6 +405,10 @@ def __eq__(self, other):
         return isinstance(other, AliasLabelExpression) and self.alias == other.alias and \
             self.expression == other.expression
 
+    def __hash__(self):
+        """Compute the hash of a AliasLabelExpression."""
```
Comment (francescofuggitti) [inline hoa2dot/core.py]: here as well

File: hoa2dot/core.py [hunk 5]
```diff
@@ -459,7 +490,10 @@ def __eq__(self, other):
 
     def __hash__(self):
         """Define hash for State."""
-        return hash((self.index, self.label, self.name, self.name, self.acc_sig))
+        try:
+            return hash((self.index, self.label, self.name, self.name, self.acc_sig))
+        except TypeError:
+            pass
```
Comment (francescofuggitti) [inline hoa2dot/core.py]: why did you add it? is there a case in which the hash could not be computed?

File: hoa2dot/parsers.py [hunk 6]
```diff
@@ -207,7 +208,8 @@ def edge(self, args):
 
     def state_conj(self, args):
         """Parse the 'state_conj' node."""
-        return args
+        # compute the flat list
+        return list(reduce(operator.add, map(lambda x: [x] if not isinstance(x, list) else x, args)))
```
Comment (francescofuggitti) [inline hoa2dot/parsers.py]: I like this pythonic way of writing code  :heart_eyes:

File: hoa2dot/core.py (hunk 5 above)
Comment (marcofavorito) [reply hoa2dot/core.py]: ops, my bad, that was added for debugging...
